File: framework/languages/node/samples/SupportChat.Ts/Server/Support/Adapters/ZLink/Spots/Handlers/set-typing-handler.ts
import { zlinkSpotActorRequestHandler } from '@zlink-systems/nestjs';
import { ConversationSpot } from '../conversation-spot';
import { SupportUserActor } from '../../Actors/support-user-actor';
import { PacketNames } from '../../../../../../Shared/Contracts/messages';
import type {
  ZLinkSpotActorRequestContext,
  ZLinkSpotActorRequestHandler
} from '@zlink-systems/framework';
import type { ConversationSpot as ConversationSpotType } from '../conversation-spot';
import type { SupportUserActor as SupportUserActorType } from '../../Actors/support-user-actor';
import type {
  SetTypingReq,
  SetTypingRes
} from '../../../../../../Shared/Contracts/messages';

@zlinkSpotActorRequestHandler({
  actor: () => SupportUserActor,
  packetName: PacketNames.setTypingReq,
  spot: () => ConversationSpot
})
class SetTypingHandler
  implements ZLinkSpotActorRequestHandler<ConversationSpotType, SupportUserActorType, SetTypingReq, SetTypingRes> {
  async handle(
    spot: ConversationSpotType,
    actor: SupportUserActorType,
    context: ZLinkSpotActorRequestContext,
    request: SetTypingReq
  ): Promise<SetTypingRes> {
    void context;
    return await spot.setTyping(actor, request);
  }
}

export { SetTypingHandler };

File: framework/languages/dotnet/samples/SupportChat/Server/Support/Infrastructure/ZLink/Spots/ConversationSpot/Handlers/SetTypingHandler.cs
using Zlink.Framework.Contracts.Handlers;
using SupportChat.Server.Support.Infrastructure.ZLink.Actors;
using SupportChat.Shared.Contracts;
using Zlink.Framework.Contracts.Spots;

namespace SupportChat.Server.Support.Infrastructure.ZLink.Spots.ConversationSpot.Handlers;

// Typing is a one-way fire-and-forget send: the sender gets no response and only
// the other participant receives a TypingChangedNotify push.
internal sealed class SetTypingHandler
    : IZLinkSpotActorSendHandler<ConversationSpot, SupportUserActor, SetTypingReq>
{
    public async ValueTask HandleAsync(
        ConversationSpot spot,
        SupportUserActor actor,
        IZLinkMessageContext context,
        SetTypingReq message,
        CancellationToken cancellationToken)
    {
        await spot.SetTypingAsync(actor, message, cancellationToken);
    }
}
